hack/set-motd/main.go

```diff
@@ -0,0 +1,165 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"time"
+)
+
+type stringFlag struct {
+	set   bool
+	value string
+}
+
+func (sf *stringFlag) Set(x string) error {
+	sf.value = x
+	sf.set = true
+	return nil
+}
+
+func (sf *stringFlag) String() string {
+	return sf.value
+}
+
+func setMotd(motd *stringFlag, user *stringFlag, pr *stringFlag, customPath *stringFlag) {
+	setHelpMsg := `
+	Usage: set-motd set [-motd string] [-user string] [-pr link]
+	  - motd: set custom message
+	  - user: set user using the server
+	  - pr:	set pull-request being tested
+	  - path: set custom path (Default: /etc/motd)
+	
+	`
+
+	if len(os.Args) == 2 || len(os.Args) > 10 {
+		fmt.Print(setHelpMsg)
+		os.Exit(1)
+	}
+
+	currentTime := time.Now()
+	var motdMsg string = fmt.Sprintf("Updated at %s\n", currentTime.Format(time.UnixDate))
+
+	if motd.set {
+		motdMsg = fmt.Sprintf("%s%s\n", motdMsg, motd.String())
+	}
+
+	if pr.set {
+		motdMsg = fmt.Sprintf("%sPull Request %s test initiated by %s\n", motdMsg, pr.String(), user.String())
+	}
+
+	if user.set && !pr.set {
+		motdMsg = fmt.Sprintf("%sUsed by %s\n", motdMsg, user.String())
+	}
+
+	// Clean file
+	var path string = customPath.String()
+	if err := os.Truncate(path, 0); err != nil {
+		log.Printf("Failed to truncate: %v", err)
+	}
+
+	// Open file
+	var f, err = os.OpenFile(path, os.O_RDWR, 0644)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	defer f.Close()
+
+	// Write motd
+	_, err = f.WriteString(motdMsg)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func unsetMotd() {
+	unsetHelpMsg := `
+	Usage: set-motd unset
+	Clears /etc/motd
+	
+	`
+
+	if len(os.Args) > 2 {
+		fmt.Print(unsetHelpMsg)
+		os.Exit(1)
+	}
+
+	// Clear file
+	if err := os.Truncate("/etc/motd", 0); err != nil {
+		log.Printf("Failed to truncate: %v", err)
+	}
+
+	// Open file
+	var f, err = os.OpenFile("/etc/motd", os.O_RDWR, 0644)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	defer f.Close()
+
+	// Write motd
+	var unsetMotd string = "Server free to use\n"
+	n, err := f.WriteString(unsetMotd)
+	if err != nil {
+		fmt.Println(err, n)
+		os.Exit(1)
+	}
+}
+
+func printHelp() {
```

Comment:
Should be used cobra as a CLI generator instead of creating this code and the main's one. 
https://github.com/spf13/cobra

Reply:
cobra was the first aim, but imho we don't want unnecessary dependencies as flag is a builtin go pkg.

Reply:
with cobra, you just use the options and you could generate a CLI reducing most all the lines you have in main.go. It's useful just to use it and create a cli in just a seconds.  Don't need to implement the code for all options that you have in main.go

But again, it's just my vision and my opinion working for a long time with go. Anyway, LGTM for internal use in hack folder. Don't want to generate a discussion for this comment.